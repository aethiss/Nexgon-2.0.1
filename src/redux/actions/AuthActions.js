import { facebookLoginManager } from '../../libs/FacebookSDK'

export function resetLogin() {
  return {
    type: 'RESET_AUTH',
  }
}

function setLogin(user) {
  return {
    type: 'SET_AUTH',
    user,
  }
}

// Handlers

export function facebookLoginRequest() {
  return dispatch => new Promise((resolve, reject) => {
    facebookLoginManager()
      .then((result) => {
        dispatch(setLogin(result))
        resolve(result)
      })
      .catch(err => reject(new Error(err)))
  })
}
