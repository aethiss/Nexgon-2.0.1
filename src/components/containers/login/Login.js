import React, { Component } from 'react'
import PropsTypes from 'prop-types'
import { connect } from 'react-redux'

// Style
import { Content, Text, View, Button } from 'native-base'

// Actions
import { loginRequest } from '../../../redux/actions/AuthActions'

// Helpers
import { facebookLoginManager, FacebookProfile } from '../../../libs/FacebookSDK'

@connect(state => ({
  authorized: state.auth.authorized,
}), { loginRequest })
export default class Login extends Component {
  static propTypes = {
    authorized: PropsTypes.bool,
    onLogin: PropsTypes.func.isRequired,
    loginRequest: PropsTypes.func,
  }

  static defaultProps = {
    authorized: false,
    loginRequest: () => {},
  }

  componentWillMount() {
    const { authorized, onLogin } = this.props
    if (authorized) onLogin()
  }

  connectToNexgon = () => {
    this.props.loginRequest()
    this.props.onLogin()
  }

  noUserRender = () =>
    (
      <View>
        <Text>No user connected !</Text>
        <Button
          iconLeft
          full
          info
          onPress={() => { facebookLoginManager() }}
        >
          <Text>Connect To NextGon</Text>
        </Button>
      </View>
    )

  render() {
    const { authorized } = this.props
    return (
      <Content>
        { !authorized && this.noUserRender() }
      </Content>
    )
  }
}
