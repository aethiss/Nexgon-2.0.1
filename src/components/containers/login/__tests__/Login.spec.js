/* eslint-disable no-undef,import/first */
import React from 'react'
import Adapter from 'enzyme-adapter-react-16'
import { shallow, configure } from 'enzyme'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import Login from '../Login'

const middlewares = [thunk]
const mockStore = configureStore(middlewares)
configure({ adapter: new Adapter() })

const initialState = {
  auth: {
    user: false,
    authorized: false,
  },
}
//
jest.mock('../../../../redux/actions/AuthActions')
import { facebookLoginRequest } from '../../../../redux/actions/AuthActions'

describe('<Login />', () => {
  // beforeEach(() => {
  //   jest.resetAllMocks()
  // })

  describe('Redender', () => {
    it('Default render', () => {
      const wrapper = shallow(
        <Login />,
        { context: { store: mockStore(initialState) } },
      )
      expect(wrapper.dive()).toMatchSnapshot()
    })
  })
  describe('Events', () => {
    it('click on connect button call facebook login', () => {
      const mocked = facebookLoginRequest.mockImplementation(() =>
        new Promise(resolve => resolve(null)))
      const wrapper = shallow(
        <Login facebookLoginRequest={facebookLoginRequest} />,
        { context: { store: mockStore(initialState) } },
      ).dive()
      wrapper.setProps({ facebookLoginRequest: mocked })
      const button = wrapper.find('.button')
      button.simulate('press')
      expect(facebookLoginRequest).toHaveBeenCalledTimes(1)
    })
  })
})
