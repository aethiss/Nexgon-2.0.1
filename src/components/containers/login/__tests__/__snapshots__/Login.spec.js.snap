// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> Redender Default render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    authorized={false}
    facebookLoginRequest={[Function]}
    onLogin={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <ImageBackground
        source={1}
        style={
          Object {
            "flex": 1,
            "height": null,
            "width": null,
          }
        }
      >
        <Styled(ViewNB)
          style={
            Object {
              "alignSelf": "center",
              "marginTop": 220,
            }
          }
        >
          <Image
            source={1}
          />
        </Styled(ViewNB)>
        <Styled(ViewNB)
          style={
            Object {
              "marginTop": 280,
              "padding": 10,
            }
          }
        >
          <Styled(Button)
            block={true}
            className="button"
            iconLeft={true}
            onPress={[Function]}
          >
            <Styled(Icon)
              name="logo-facebook"
            />
            <Styled(Text)>
              Connect To NextGon!
            </Styled(Text)>
          </Styled(Button)>
        </Styled(ViewNB)>
        <Styled(ViewNB)
          style={
            Object {
              "alignSelf": "center",
            }
          }
        >
          <Styled(Text)
            style={
              Object {
                "color": "#FFF",
              }
            }
          >
            Login with Facebook
          </Styled(Text)>
        </Styled(ViewNB)>
      </ImageBackground>,
      "contentContainerStyle": Object {
        "flex": 1,
        "flexDirection": "column",
      },
      "virtual": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(ViewNB)
            style={
              Object {
                "alignSelf": "center",
                "marginTop": 220,
              }
            }
          >
            <Image
              source={1}
            />
          </Styled(ViewNB)>,
          <Styled(ViewNB)
            style={
              Object {
                "marginTop": 280,
                "padding": 10,
              }
            }
          >
            <Styled(Button)
              block={true}
              className="button"
              iconLeft={true}
              onPress={[Function]}
            >
              <Styled(Icon)
                name="logo-facebook"
              />
              <Styled(Text)>
                Connect To NextGon!
              </Styled(Text)>
            </Styled(Button)>
          </Styled(ViewNB)>,
          <Styled(ViewNB)
            style={
              Object {
                "alignSelf": "center",
              }
            }
          >
            <Styled(Text)
              style={
                Object {
                  "color": "#FFF",
                }
              }
            >
              Login with Facebook
            </Styled(Text)>
          </Styled(ViewNB)>,
        ],
        "source": 1,
        "style": Object {
          "flex": 1,
          "height": null,
          "width": null,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Image
              source={1}
            />,
            "style": Object {
              "alignSelf": "center",
              "marginTop": 220,
            },
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "source": 1,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Styled(Button)
              block={true}
              className="button"
              iconLeft={true}
              onPress={[Function]}
            >
              <Styled(Icon)
                name="logo-facebook"
              />
              <Styled(Text)>
                Connect To NextGon!
              </Styled(Text)>
            </Styled(Button)>,
            "style": Object {
              "marginTop": 280,
              "padding": 10,
            },
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "block": true,
              "children": Array [
                <Styled(Icon)
                  name="logo-facebook"
                />,
                <Styled(Text)>
                  Connect To NextGon!
                </Styled(Text)>,
              ],
              "className": "button",
              "iconLeft": true,
              "onPress": [Function],
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "name": "logo-facebook",
                  "virtual": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Connect To NextGon!",
                  "virtual": undefined,
                },
                "ref": null,
                "rendered": "Connect To NextGon!",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Styled(Text)
              style={
                Object {
                  "color": "#FFF",
                }
              }
            >
              Login with Facebook
            </Styled(Text)>,
            "style": Object {
              "alignSelf": "center",
            },
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Login with Facebook",
              "style": Object {
                "color": "#FFF",
              },
              "virtual": undefined,
            },
            "ref": null,
            "rendered": "Login with Facebook",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <ImageBackground
          source={1}
          style={
            Object {
              "flex": 1,
              "height": null,
              "width": null,
            }
          }
        >
          <Styled(ViewNB)
            style={
              Object {
                "alignSelf": "center",
                "marginTop": 220,
              }
            }
          >
            <Image
              source={1}
            />
          </Styled(ViewNB)>
          <Styled(ViewNB)
            style={
              Object {
                "marginTop": 280,
                "padding": 10,
              }
            }
          >
            <Styled(Button)
              block={true}
              className="button"
              iconLeft={true}
              onPress={[Function]}
            >
              <Styled(Icon)
                name="logo-facebook"
              />
              <Styled(Text)>
                Connect To NextGon!
              </Styled(Text)>
            </Styled(Button)>
          </Styled(ViewNB)>
          <Styled(ViewNB)
            style={
              Object {
                "alignSelf": "center",
              }
            }
          >
            <Styled(Text)
              style={
                Object {
                  "color": "#FFF",
                }
              }
            >
              Login with Facebook
            </Styled(Text)>
          </Styled(ViewNB)>
        </ImageBackground>,
        "contentContainerStyle": Object {
          "flex": 1,
          "flexDirection": "column",
        },
        "virtual": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Styled(ViewNB)
              style={
                Object {
                  "alignSelf": "center",
                  "marginTop": 220,
                }
              }
            >
              <Image
                source={1}
              />
            </Styled(ViewNB)>,
            <Styled(ViewNB)
              style={
                Object {
                  "marginTop": 280,
                  "padding": 10,
                }
              }
            >
              <Styled(Button)
                block={true}
                className="button"
                iconLeft={true}
                onPress={[Function]}
              >
                <Styled(Icon)
                  name="logo-facebook"
                />
                <Styled(Text)>
                  Connect To NextGon!
                </Styled(Text)>
              </Styled(Button)>
            </Styled(ViewNB)>,
            <Styled(ViewNB)
              style={
                Object {
                  "alignSelf": "center",
                }
              }
            >
              <Styled(Text)
                style={
                  Object {
                    "color": "#FFF",
                  }
                }
              >
                Login with Facebook
              </Styled(Text)>
            </Styled(ViewNB)>,
          ],
          "source": 1,
          "style": Object {
            "flex": 1,
            "height": null,
            "width": null,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Image
                source={1}
              />,
              "style": Object {
                "alignSelf": "center",
                "marginTop": 220,
              },
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "source": 1,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Styled(Button)
                block={true}
                className="button"
                iconLeft={true}
                onPress={[Function]}
              >
                <Styled(Icon)
                  name="logo-facebook"
                />
                <Styled(Text)>
                  Connect To NextGon!
                </Styled(Text)>
              </Styled(Button)>,
              "style": Object {
                "marginTop": 280,
                "padding": 10,
              },
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "block": true,
                "children": Array [
                  <Styled(Icon)
                    name="logo-facebook"
                  />,
                  <Styled(Text)>
                    Connect To NextGon!
                  </Styled(Text)>,
                ],
                "className": "button",
                "iconLeft": true,
                "onPress": [Function],
                "virtual": undefined,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "name": "logo-facebook",
                    "virtual": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Connect To NextGon!",
                    "virtual": undefined,
                  },
                  "ref": null,
                  "rendered": "Connect To NextGon!",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Styled(Text)
                style={
                  Object {
                    "color": "#FFF",
                  }
                }
              >
                Login with Facebook
              </Styled(Text)>,
              "style": Object {
                "alignSelf": "center",
              },
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Login with Facebook",
                "style": Object {
                  "color": "#FFF",
                },
                "virtual": undefined,
              },
              "ref": null,
              "rendered": "Login with Facebook",
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
